src/functions/validator/checkEmailExist.ts

```diff
@@ -0,0 +1,27 @@
+import { SubjectWithoutIdentifierError } from 'typeorm';
+import { AppDataSource } from '../../data-source';
+import { User } from '../../entity/user';
+import { GraphQLError } from 'graphql';
+
+const userRepo = AppDataSource.getRepository(User);
+
+export const checkEmail = async (email: string) => {
+  const emailCheck = await userRepo.findOne({
+    select: {
+      email: true,
+    },
+    where: {
+      email: email,
+    },
+  });
+
+  if (emailCheck == null) {
+    return true;
```

Comment:
precisa retornar `true`?

Reply:
também acho que é sempre melhor usar [strict equality](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality), embora entenda que você deve estar pensando em casos onde poderia ser qualquer valor `falsy`, mas é mais seguro usar [double not](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_NOT#double_not_!!)